feature-pack-api/src/test/java/org/jboss/provisioning/state/xml/test/ProvisionedStateXmlWriterTestCase.java

```diff
@@ -0,0 +1,115 @@
+package org.jboss.provisioning.state.xml.test;
+
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.logging.Logger;
+
+import org.xml.sax.SAXException;
+
+import org.jboss.provisioning.ArtifactCoords;
+import org.jboss.provisioning.runtime.ResolvedFeatureId;
+import org.jboss.provisioning.runtime.ResolvedSpecId;
+import org.jboss.provisioning.state.ProvisionedFeaturePack;
+import org.jboss.provisioning.state.ProvisionedPackage;
+import org.jboss.provisioning.state.ProvisionedState;
+import org.jboss.provisioning.test.util.XmlParserValidator;
+import org.jboss.provisioning.xml.ProvisionedConfigBuilder;
+import org.jboss.provisioning.xml.ProvisionedFeatureBuilder;
+import org.jboss.provisioning.xml.ProvisionedStateXmlParser;
+import org.jboss.provisioning.xml.ProvisionedStateXmlWriter;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Tomas Hofman (thofman@redhat.com)
+ */
+public class ProvisionedStateXmlWriterTestCase {
+
+    private static final String SCHEMA = "schema/pm-provisioned-state-1_0.xsd";
+
+    private Logger log = Logger.getLogger(getClass().getName());
+    private XmlParserValidator<ProvisionedState> validator;
+
+    @Before
+    public void before() throws URISyntaxException {
+        URL xsd = getClass().getClassLoader().getResource(SCHEMA);
+        Assert.assertNotNull(xsd);
+        validator = new XmlParserValidator<>(Paths.get(xsd.toURI()), ProvisionedStateXmlParser.getInstance());
+    }
+
+    @Test
+    public void testMarshallUnmarshall() throws Exception {
+        ProvisionedState originalState = ProvisionedState.builder()
+                .addFeaturePack(ProvisionedFeaturePack.builder(ArtifactCoords.newGav("org.jboss.group1", "fp1", "1.0.0"))
+                        .addPackage("package1")
+                        .addPackage(ProvisionedPackage.newInstance("package2"))
+                        .build())
+                .addFeaturePack(ProvisionedFeaturePack.builder(ArtifactCoords.newGav("org.jboss.group1", "fp2", "1.0.0"))
+                        .addPackage("package3")
+                        .build())
+                // TODO: Use ProvisionedConfigBuilder or ConfigModelBuilder to create configs?
```

Comment:
ProvisionedConfigBuilder. ConfigModelBuilder is used by the mechanism and is not usable otherwise.